openedx/core/djangoapps/programs/utils.py

```diff
@@ -105,3 +74,32 @@ def get_programs_for_dashboard(user, course_keys):
             log.exception('Unable to parse Programs API response: %r', program)
 
     return course_programs
+
+
+def get_programs_for_credentials(user, programs_credentials):
+    """ Given a user and an iterable of credentials, get corresponding programs
+    data and return it as a list of dictionaries.
+
+    Arguments:
+        user (User): The user to authenticate as for requesting programs.
+        programs_credentials (list): List of credentials awarded to the user
+            for completion of a program.
+
+    Returns:
+        list, containing programs dictionaries.
+    """
+    ProgramsApiConfig.current()
+    certificate_programs = []
+
+    programs = get_programs(user)
+    if not programs:
+        log.debug('No programs found for the user with ID %d.', user.id)
+        return certificate_programs
+
+    for program in programs:
+        for credential in programs_credentials:
```

Comment:
Use [`groupby`](https://docs.python.org/2/library/itertools.html#itertools.groupby) to map the credentials by program ID.